src/ripple/app/ledger/tests/Ledger_test.cpp

```diff
@@ -0,0 +1,149 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tests/common_ledger.h>
+
+namespace ripple {
+namespace test {
+
+class Ledger_test : public beast::unit_test::suite
+{
+    void test_genesisLedger (bool sign)
+    {
+        std::uint64_t const xrp = std::mega::num;
+
+        auto master = createAccount ("masterpassphrase");
+
+        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
+
+        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+
+        // User accounts
+        auto gw1 = createAccount ("gw1");
+        expect (gw1.pk != master.pk, "gw1.pk != master.pk");
```

Comment:
Well, `let me remind you` tests and code don't arrive bug free. We're talking about process vs product here. Besides which, my point is we can just wrap the expression in a MACRO, and get a bunch of extra information for next to nothing. It's not bloat at all. The constructs provide facility to remove repetitious code, which keeps popping up.

```expect (gw1.pk != master.pk, "gw1.pk != master.pk");```

vs 

```EXPECT (gw1.pk != master.pk);```

Which is more bloated?

Here's another specimen from the m-of-n branch:

```
+            expect (serialized == expectPass,

+                "Unexpected serialized = " + std::to_string (serialized) +

+                      ".  Object:\n" + signingAcct.getFullText () + "\n");
```

People often use tests to debug, often writing them WHILE they are developing. You can ignore this FACT, and insist people use debuggers, so you can maintain the `elegance` of the test suite or you can provide useful facilities to formalise a lot of this scaffolding into some thing of lasting utility.